File: Examino/Server/Examino.Domain/Entities/Raport.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Examino.Domain.Entities
{
    public class Raport
    {
        public Guid Id { get; set; }
        [ForeignKey("Patient")]
        public Guid? PatientId { get; set; }
        [ForeignKey("Doctor")]
        public Guid? DoctorId { get; set; }
        public DateTimeOffset RaportTime { get; set; }
        public string Symptoms { get; set; }
        public string Examination { get; set; }
        public string Diagnosis { get; set; }
        public string Recommendation { get; set; }
        public string Comment { get; set; }

        public virtual Prescription? Prescription { get; set; }
        public virtual Patient? Patient { get; set; }
        public virtual Doctor?  Doctor { get; set; }
    }
}

File: Examino/App/src/models/Raport.ts
export interface Raport {
  patientName: string;
  patientSurname: string;
  doctorName: string;
  doctorSurname: string;
  raport: {
    id: string;
    patientId: string;
    doctorId: string;
    raportTime: Date;
    symptoms: string;
    examination: string;
    diagnosis: string;
    recommendation: string;
    comment: string;
  };
  prescription: {
    id: string;
    raportId: string;
    medicines: string;
  };
}

export interface CreateRaport {
  patientId: string;
  doctorId: string;
  symptoms: string;
  examination: string;
  diagnosis: string;
  recommendation: string;
  comment: string;
}
